src/bosh_azure_cpi/lib/cloud/azure/restapi/vmss_rest.rb

```diff
@@ -0,0 +1,318 @@
+
+    # Instances related starts
+    def get_vmss_instances(resource_group_name, name)
+      url = rest_api_url(REST_API_PROVIDER_COMPUTE, REST_API_VIRTUAL_MACHINE_SCALE_SETS, resource_group_name: resource_group_name, name: name)
+      url += '/virtualMachines'
+      result = get_resource_by_id(url)
+      vmss_instances = nil
+      unless result.nil?
+        vmss_instances = []
+        result['value'].each do |instance|
+          vmss_instances.push(
+            instanceId: instance['instanceId'],
+            name: instance['name'],
+            zones: instance['zones']
+          )
+        end
+      end
+      vmss_instances
+    end
+
+    def get_vmss_instance(resource_group_name, vmss_name, vmss_instance_id)
+      vmss_instance = nil
+      result = _get_vmss_instance(resource_group_name, vmss_name, vmss_instance_id)
+      unless result.nil?
+        vmss_instance = {}
+        vmss_instance[:id]          = result['id']
+        vmss_instance[:name]        = result['name']
+        vmss_instance[:location]    = result['location']
+        vmss_instance[:sku]         = {}
+        vmss_instance[:sku][:name]  = result['sku']['name']
+        vmss_instance[:zones]       = result['zones']
+        vmss_instance[:tags]        = result['tags']
+        properties = result['properties']
+        vmss_instance[:provisioning_state] = properties['provisioningState']
+        storage_profile = properties['storageProfile']
+        os_disk = storage_profile['osDisk']
+        vmss_instance[:os_disk] = {}
+        vmss_instance[:os_disk][:name]    = os_disk['name']
+        vmss_instance[:os_disk][:caching] = os_disk['caching']
+        vmss_instance[:os_disk][:size]    = os_disk['diskSizeGb']
+
+        vmss_instance[:os_disk][:uri]     = os_disk['vhd']['uri'] if os_disk.key?('vhd')
+        if os_disk.key?('managedDisk')
+          vmss_instance[:os_disk][:managed_disk] = {}
+          vmss_instance[:os_disk][:managed_disk][:id]                   = os_disk['managedDisk']['id']
+          vmss_instance[:os_disk][:managed_disk][:storage_account_type] = os_disk['managedDisk']['storageAccountType']
+        end
+
+        vmss_instance[:data_disks] = []
+        storage_profile['dataDisks'].each do |data_disk|
+          disk = {}
+          disk[:name]    = data_disk['name']
+          disk[:lun]     = data_disk['lun']
+          disk[:caching] = data_disk['caching']
+          disk[:size]    = data_disk['diskSizeGb']
+
+          disk[:uri]     = data_disk['vhd']['uri'] if data_disk.key?('vhd')
+          if data_disk.key?('managedDisk')
+            disk[:managed_disk] = {}
+            disk[:managed_disk][:id]                   = data_disk['managedDisk']['id']
+            disk[:managed_disk][:storage_account_type] = data_disk['managedDisk']['storageAccountType']
+          end
+
+          # TODO: handle this.
+          # disk[:disk_bosh_id] = result['tags'].fetch("#{DISK_ID_TAG_PREFIX}-#{data_disk['name']}", data_disk['name'])
```

Comment:
This will break the CPI method `get_disks` and the scenario of `bosh cck`.

Reply:
yes, that's why put one TODO here. 
will find one place to store the tags, I think maybe the 'meta store' i ever thought of.

Reply:
and also the tags for the vm seems would be added in Vanadium